integration/integration_test.go

```diff
@@ -185,439 +172,296 @@ func getIntegrationTest(numNodes, numVotes int) func(*testing.T) {
+	return func(t *testing.T) {
+		t.Parallel()
 
-		ctx, cancel := context.WithTimeout(context.Background(), m.t)
-		defer cancel()
+		adminID := "first admin"
 
-		events := m.n.GetOrdering().Watch(ctx)
+		// ##### SETUP ENV #####
+		// make tests reproducible
+		rand.Seed(1)
 
-		err = m.n.GetPool().Add(sentTxn)
-		if err != nil {
-			return nil, xerrors.Errorf("failed to Add: %v", err)
-		}
+		delaPkg.Logger = delaPkg.Logger.Level(zerolog.WarnLevel)
 
-		sentTxnID := sentTxn.GetID()
+		dirPath, err := os.MkdirTemp(os.TempDir(), "d-voting-three-votes")
+		require.NoError(t, err)
 
-		accepted := isAccepted(events, sentTxnID)
-		if accepted {
-			return sentTxnID, nil
-		}
+		defer os.RemoveAll(dirPath)
 
-		err = m.m.Sync()
-		if err != nil {
-			return nil, xerrors.Errorf("failed to sync: %v", err)
-		}
+		t.Logf("using temp dir %s", dirPath)
 
-		cancel()
-	}
+		// ##### CREATE NODES #####
+		nodes := setupDVotingNodes(t, numNodes, dirPath)
 
-	return nil, xerrors.Errorf("transaction not included after timeout: %v", args)
-}
+		signer := createDVotingAccess(t, nodes, dirPath)
 
-// isAccepted returns true if the transaction was included then accepted
-func isAccepted(events <-chan ordering.Event, txID []byte) bool {
-	for event := range events {
-		for _, result := range event.Transactions {
-			fetchedTxnID := result.GetTransaction().GetID()
+		m := newTxManager(signer, nodes[0], time.Second*time.Duration(numNodes/2+1), numNodes*4)
 
-			if bytes.Equal(txID, fetchedTxnID) {
-				accepted, _ := event.Transactions[0].GetStatus()
+		err = grantAccess(m, signer)
+		require.NoError(t, err)
 
-				return accepted
-			}
+		for _, n := range nodes {
+			err = grantAccess(m, n.GetShuffleSigner())
+			require.NoError(t, err)
 		}
-	}
-
-	return false
-}
 
-func grantAccess(m txManager, signer crypto.Signer) error {
-	pubKeyBuf, err := signer.GetPublicKey().MarshalBinary()
-	if err != nil {
-		return xerrors.Errorf("failed to GetPublicKey: %v", err)
-	}
-
-	args := []txn.Arg{
-		{Key: native.ContractArg, Value: []byte("go.dedis.ch/dela.Access")},
-		{Key: "access:grant_id", Value: []byte(hex.EncodeToString(evotingAccessKey[:]))},
-		{Key: "access:grant_contract", Value: []byte("go.dedis.ch/dela.Evoting")},
-		{Key: "access:grant_command", Value: []byte("all")},
-		{Key: "access:identity", Value: []byte(base64.StdEncoding.EncodeToString(pubKeyBuf))},
-		{Key: "access:command", Value: []byte("GRANT")},
-	}
-	_, err = m.addAndWait(args...)
-	if err != nil {
-		return xerrors.Errorf("failed to grantAccess: %v", err)
-	}
-
-	return nil
-}
-
-func createForm(m txManager, title string, admin string) ([]byte, error) {
-	// Define the configuration :
-	configuration := fake.BasicConfiguration
+		// ##### CREATE FORM #####
+		formID, err := createForm(m, "Three votes form", adminID)
+		require.NoError(t, err)
 
-	createForm := types.CreateForm{
-		Configuration: configuration,
-		AdminID:       admin,
-	}
+		time.Sleep(time.Second * 1)
```

Comment:
Have you tried without the sleep? The function createForm waits and checks that the tx is accepted so it makes me think it shouldn't be necessary